lte/gateway/c/oai/tasks/mme_app/mme_app_context.c

```diff
@@ -1972,8 +1972,7 @@ void mme_app_handle_s1ap_ue_context_release_complete(
   mme_app_delete_s11_procedure_create_bearer(ue_context_p);
 
   if (ue_context_p->mm_state == UE_UNREGISTERED) {
-    if ((ue_context_p->mme_teid_s11 == 0) &&
-        (!ue_context_p->nb_active_pdn_contexts)) {
+    if (!ue_context_p->nb_active_pdn_contexts) {
```

Comment:
explicitly state `ue_context_p->nb_active_pdn_contexts > 0`.

Reply:
why is `(ue_context_p->mme_teid_s11 == 0)` no longer checked (cc @pruthvihebbani)?

Reply:
In previous code, on reception of Delete Session Response, mme_teid_s11 was set to "0" then it would send Ue Context Release Request; On reception of UE Context Release Complete, if UE's emm state is un-registered then mme would free up all contexts associated to that UE; Since mme_teid_s11 was set to zero; it  was not able to remove mme_teid_s11 from hashlist, remove tun11_ue_context_htbl.

The parameter nb_active_pdn_contexts indicates the number of active PDNs; if there are no active PDNs and UE's state is de-registered state, then mme should release all contexts associated to the UE

Reply:
> explicitly state `ue_context_p->nb_active_pdn_contexts > 0`.

Condition validates that if UE doesn't have any active pdn contexts then it should proceed to delete UE context;
So we can correct the statement as if (ue_context_p->nb_active_pdn_contexts == 0)

Reply:
indeed. thanks for catching the bug in my comment :)